modules/kafka/src/main/java/com/devonfw/module/kafka/common/messaging/util/MessageUtil.java

```diff
@@ -0,0 +1,49 @@
+package com.devonfw.module.kafka.common.messaging.util;
+
+import java.nio.charset.Charset;
+
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+
+/**
+ * @author ravicm
+ *
+ */
+public class MessageUtil {
```

Comment:
Please use kafkas native headers.

Reply:
Umm .. Where exactly ? i mean the method parameter of addHeader. the Headers is already from 
org.apache.kafka.common.header.Headers

Reply:
Maybe I was to quick here. I saw that MessageParserImpl#parse extracts the header froms the payload. MessageUtil itself is fine.